apps/speed.c

```diff
@@ -1916,6 +1933,32 @@ int speed_main(int argc, char **argv)
             }
         }
     }
+    /* repeated code good to factorize */
+    ecdh_c[R_EC_BRP256R1][0] = count / 1000;
+    for (i = R_EC_BRP384R1; i < R_EC_BRP512R1; ++i) {
+        ecdh_c[i][0] = ecdh_c[i - 1][0] / 2;
+        if (ecdh_doit[i] <= 1 && ecdh_c[i][0] == 0)
+            ecdh_doit[i] = 0;
+        else {
+            if (ecdh_c[i][0] == 0) {
+                ecdh_c[i][0] = 1;
+            }
+        }
+    }
+    ecdh_c[R_EC_BRP256T1][0] = count / 1000;
+    for (i = R_EC_BRP384T1; i < R_EC_BRP512T1; ++i) {
+        ecdh_c[i][0] = ecdh_c[i - 1][0] / 2;
+        if (ecdh_doit[i] <= 1 && ecdh_c[i][0] == 0)
+            ecdh_doit[i] = 0;
+        else {
+            if (ecdh_c[i][0] == 0) {
+                ecdh_c[i][0] = 1;
+            }
+        }
+    }
+    /* default iteration count for the last two EC Curves */
+    ecdh_c[R_EC_X25519][0] = ecdh_c[R_EC_X448][0] = count / 1000;
```

Comment:
Hmm, this is not right. Basically the difference between these values is approximate difference between speeds, so I'd suggest count / 8000 for X448.

Reply:
The brp numbers are unreasonable too. The brp256 should be ~3-5 smaller than they are, then loop condition is totally wrong for chosen indices. Indeed, the R and T are interleaved, so it should be i+=2, and it should be <= R_EC_BRP512[RT]1...

Reply:
I change my mind twice about interleaving or not : that's why this change is awkward.
You know that this  juggle around estimating iteration counts is mostly useless, because SIGALRM define and support is the main use-case ...

Reply:
If you interleave, then it's i+=2, if you ++i, then there should be no interleave. If argument is that it's not non-main use case, then it's argument in favour of whole thing removal, **not** for suggesting code that is visibly (and verifiably) wrong. There is no excuse for that.